improve system-variables documentation (#4598)
pingcap/docs: system-variables.md

## Variable Reference

### allow_auto_random_explicit_insert <span class="version-mark">New in v4.0.3</span>

- Scope: SESSION (since v4.0.5: SESSION | GLOBAL)
- Default value: OFF
- Determines whether to allow explicitly specifying the values of the column with the `AUTO_RANDOM` attribute in the `INSERT` statement.

### auto_increment_increment

- Scope: SESSION | GLOBAL
- Default value: 1

> - Controls how `AUTO_INCREMENT` values should be incremented on each allocation. Often used in combination with `auto_increment_offset`.

Reviewer: ```suggestion
- Controls how `AUTO_INCREMENT` values should be incremented on each allocation. It is often used in combination with `auto_increment_offset`.
```

>
> Unlike in MySQL, the `max_execution_time` system variable currently works on all kinds of statements in TiDB, not only restricted to the `SELECT` statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specify.

### port

- Scope: NONE
- Default value: 4000

> - The port that the `tidb-server` is listening on when speaking the MySQL protocol.

Reviewer: What does "speaking" mean?

Author: TiDB has a HTTP (10080) and a MySQL port (4000). Clients are said to "talk to the server", the network protocol the server speaks is "MySQL". This is different from SQL, which happens at a higher level. Talking/speaking is the basic protocol negotiation.

### tidb_capture_plan_baselines <span class="version-mark">New in v4.0</span>

- Scope: SESSION | GLOBAL
- Default value: OFF
- This variable is used to control whether to enable the [baseline capturing](/sql-plan-management.md#baseline-capturing) feature. This feature depends on the statement summary, so you need to enable the statement summary before you use baseline capturing.
- After this feature is enabled, the historical SQL statements in the statement summary are traversed periodically, and bindings are automatically created for SQL statements that appear at least twice.

### tidb_check_mb4_value_in_utf8

- Scope: INSTANCE
- Default value: ON
- This variable is used to enforce that the `utf8` character set only stores values from the [Basic Multilingual Plane (BMP)](https://en.wikipedia.org/wiki/Plane_\(Unicode\)#Basic_Multilingual_Plane). To store characters outside the BMP, it is recommended to use the `utf8mb4` character set.

> - You may need to disable this option when upgrading from an earlier version of TiDB, where `utf8` checking was more relaxed. For details, see [FAQs After Upgrade](/faq/upgrade-faq.md).

Reviewer: ```suggestion
- You might need to disable this option when upgrading your cluster from an earlier version of TiDB where the `utf8` checking was more relaxed. For details, see [FAQs After Upgrade](/faq/upgrade-faq.md).
```

### tidb_enable_slow_log

- Scope: INSTANCE
- Default value: ON
- This variable is used to control whether to enable the slow log feature.

### tidb_enable_stmt_summary <span class="version-mark">New in v3.0.4</span>

- Scope: SESSION | GLOBAL
- Default value: ON (the value of the default configuration file)
- This variable is used to control whether to enable the statement summary feature. If enabled, SQL execution information like time consumption is recorded to the `information_schema.STATEMENTS_SUMMARY` system table to identify and troubleshoot SQL performance issues.

### tidb_enable_table_partition

- Scope: SESSION | GLOBAL
- Default value: ON
- This variable is used to set whether to enable the `TABLE PARTITION` feature:

    - `ON` indicates enabling Range partitioning, Hash partitioning, and Range column partitioning with one single column.
    - `AUTO` functions the same way as `ON` does.
    - `NIGHTLY` indicates enabling the types of partitioned tables specified in `ON`. It also indicates enabling List partitioning and List COLUMNS partitioning.

>     - `OFF` indicates disabling the `TABLE PARTITION` feature. In this case, the syntax that creates a partition table can be executed, but the table created is not a partitioned one.

Reviewer: List columns feature is not released with TiDB 5.0 RC. So this doc change is reverted in https://github.com/pingcap/docs/pull/4593/files#diff-ec2cbb115cc58fe3afe2862eb5148732d323cbf7ae35dea97bd381a1f8d0e8a0R374-R378.

Author: Can we merge to master and then remove from 5.0-rc?

Reviewer: The content of List partition/List columns partition is also removed from `master`. 
```suggestion
    - `ON` indicates enabling Range partitioning, Hash partitioning, and Range column partitioning with one single column.
    - `AUTO` functions the same way as `ON` does.
    - `OFF` indicates disabling the `TABLE PARTITION` feature. In this case, the syntax that creates a partition table can be executed, but the table created is not a partitioned one.
```

- Scope: SESSION | GLOBAL
- Default value: -1
- This variable is used to set the concurrency of the `hash join` algorithm.

> - A value of `-1` means that the value of `tidb_executor_concurrency` will be used instead.

Reviewer: @morgo @XuHuaiyu 
Shall we add a note or warning to indicate it's deprecated in 5.0? For example: https://github.com/pingcap/docs/pull/4602/files#diff-ec2cbb115cc58fe3afe2862eb5148732d323cbf7ae35dea97bd381a1f8d0e8a0R458-R460

Author: Yes. It is the same case as tidb_hashagg_final_concurrency.

Author: I checked the file. There is already a depecation warning.

    select * from t, t1 where t.a=t1.a
    ```

### tidb_opt_prefer_range_scan

- Scope: SESSION
- Default value: 0
- After you set the value of this variable to `1`, the optimizer always prefers index scans over full table scans.

> - In the following example, before you enable `tidb_opt_prefer_range_scan`, the TiDB optimizer performs a full table scan . After you enable `tidb_opt_prefer_range_scan`, the optimizer selects an index scan.

Reviewer 2: ```suggestion
- In the following example, before you enable `tidb_opt_prefer_range_scan`, the TiDB optimizer performs a full table scan. After you enable `tidb_opt_prefer_range_scan`, the optimizer selects an index scan.
```

+-------------------------+------------+-----------+---------------+-------------------+
3 rows in set (0.00 sec)

set session tidb_opt_prefer_range_scan = 1;

explain select * from t where age=5;
+-------------------------------+------------+-----------+-----------------------------+-------------------------------+
| id                            | estRows    | task      | access object               | operator info                 |
+-------------------------------+------------+-----------+-----------------------------+-------------------------------+
| IndexLookUp_7                 | 1048576.00 | root      |                             |                               |
| ├─IndexRangeScan_5(Build)     | 1048576.00 | cop[tikv] | table:t, index:idx_age(age) | range:[5,5], keep order:false |
| └─TableRowIDScan_6(Probe)     | 1048576.00 | cop[tikv] | table:t                     | keep order:false              |
+-------------------------------+------------+-----------+-----------------------------+-------------------------------+
3 rows in set (0.00 sec)
```

### tidb_opt_write_row_id

- Scope: SESSION
- Default value: OFF

> - This variable is used to set whether to allow `INSERT`, `REPLACE` and `UPDATE` statements to operate on the column `_tidb_rowid`. It is not allowed by default. This variable can be used only when importing data with TiDB tools.

Reviewer 2: ```suggestion
- This variable is used to set whether to allow `INSERT`, `REPLACE` and `UPDATE` statements to operate on the column `_tidb_rowid`. This variable can be used only when importing data with TiDB tools.
```
